docs/development/adr/0018-bin-hist-reduction-dims.md

```diff
@@ -0,0 +1,123 @@
+# ADR 0018: More explicit control of replaced dims in `bin` and `hist` functions
+
+-   Status: proposed
+-   Deciders: Simon, Jan-Lukas
+-   Date: 2024-07-12
+
+## Context
+
+### Background
+
+Currently functions such as `bin` and `hist` derive their behavior from looking at whether the coordinate being used for the operation is a "dimension-coordinate" of the data array, i.e., a coordinate with its name matching a dimension of the data.
+Recently we have seen more and more cases where this behavior leads to surprising and too implicit results.
+To control the actual behavior a number of workarounds in user-code are required, using combinations of functions such as `flatten`, `group`, `rename_dims`, `drop_coords`, `transpose`, etc.
+Furthermore, there are cases where the default behavior shows sub-optimal performance, which can be avoided using further workarounds.
+All of this is cumbersome and error prone, so this ADR proposes a more predictable and explicit control of the dimension(s) to be replaced or added.
+
+### Proposed solution
+
+1. Add a `dim` argument to all `bin` and `hist` functions to specify the dimension(s) to be replaced.
+2. The default `dim=None` means the dimension of the *coordinate(s)* used for the operation.
+3. In case of binned data there can be cases without coordinate, since only the bin contents might have to relevant coordinate.
```

Comment:
```suggestion
3. In case of binned data there can be cases without coordinate, since only the bin contents might have the relevant coordinate.
```
...?